core/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -516,6 +522,42 @@ public void testFromXContentWithHeadersAndMetadata() throws IOException {
         assertThat(cause.getMetadata("es.index_uuid"), hasItem("_na_"));
     }
 
+    public void testThrowableToAndFromXContent() throws IOException {
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+
+        final Tuple<Throwable, ElasticsearchException> exceptions = randomExceptions();
+        final Throwable throwable = exceptions.v1();
+
+        BytesReference throwableBytes = XContentHelper.toXContent((builder, params) -> {
+            ElasticsearchException.generateThrowableXContent(builder, params, throwable);
+            return builder;
+        }, xContent.type(), randomBoolean());
+
+        ElasticsearchException parsedException;
+        try (XContentParser parser = createParser(xContent, throwableBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            parsedException = ElasticsearchException.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertNull(parser.nextToken());
+        }
+        assertNotNull(parsedException);
+
+        ElasticsearchException expected = exceptions.v2();
+        while (expected != null) {
```

Comment:
nit: here I would maybe use do while just to make absolutely sure that the first round always runs.